### ISSUE TITLE
Discrepancy between rule descriptions and implementations in `comment-without-dot`, `incorrect-version`, and `not-empty-atom` lints

### ISSUE THREAD

**Issue by VasilevNStas** [objectionary/lints#1384]
## Problem

In three lints the defect text promises a rule that the XPath condition does not implement (or vice versa).

## 1. `comment-without-dot` — text allows `:`, condition does not

`src/main/resources/org/eolang/lints/comments/comment-without-dot.xsl:27`:

> "The comment for the object doesn't end with a dot (or '?', '!', ':')"

but the condition (`:13`) only checks `.`, `?`, `!`:

```xml
<xsl:for-each select="/object/comments/comment[not(ends-with(., '.')) and not(ends-with(., '?')) and not(ends-with(., '!'))]">
```

A comment ending with `:` is flagged although the message explicitly permits it.

## 2. `incorrect-version` — text promises strict SemVer, regex allows `-SNAPSHOT`

`src/main/resources/org/eolang/lints/metas/incorrect-version.xsl:16` (regex) vs `:32` (message "SemVer expected"). The regex additionally accepts `1.0-SNAPSHOT`, which is not a SemVer version — a user is told SemVer is required, but a non-SemVer value passes.

## 3. `not-empty-atom` — message lists void attributes the rule excludes

`src/main/resources/org/eolang/lints/atoms/not-empty-atom.xsl:31` builds the message from `o[@base and not(@name='λ')]` (which includes void attributes `@base='∅'`), while the selector (`:14`) excludes them (`@base!='∅'`). For an atom with both a void and a non-void attribute, the message names the void attribute as the violation.

## Expected behavior

Each lint's message and its condition must describe the same rule (align the condition with the text, or vice versa), and add packs for the boundary cases.

## Related

- `funcs/escape.xsl` — used by these lints to quote names in messages.

**Comment by 0crat** [comment]
@vasilevnstas Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+830](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
